docs/reference.md

```diff
@@ -1971,6 +1971,165 @@ str_field:
+
+```q
+q)table:([] i8_fd:5?0x64; i16_fd:5?100h; i32_fd:5?100i)
+q).arrowkdb.orc.writeOrcFromTable["dataloader.orc";table;::]
+```
+
+### `orc.readOrcSchema`
+
+*Read the schema from an Apache ORC file*
+
+```txt
+.arrowkdb.orc.readOrcSchema[orc_file]
+```
+
+Where `orc_file` is a string containing the ORC file name
+
+returns the schema identifier
+
+> :warning: The Apache ORC file format is not supported on Windows platform and will return with the message **ORC files are not supported on Windows**.
+
+```q
+q)i8_fd:.arrowkdb.fd.field[`int8;.arrowkdb.dt.int8[]];
+q)i16_fd:.arrowkdb.fd.field[`int16;.arrowkdb.dt.int16[]];
+q)i32_fd:.arrowkdb.fd.field[`int32;.arrowkdb.dt.int32[]];
+q)orc_schema:.arrowkdb.sc.schema[(i8_fd,i16_fd,i32_fd)];
+q)orc_data:(5?0x64;5?100h;5?100i);
+q).arrowkdb.orc.writeOrc["dataloader.orc";orc_schema;orc_data;options]
+q).arrowkdb.sc.equalSchemas[schema;.arrowkdb.orc.readOrcSchema["dataloader.orc"]]
+1b
+```
+
+### `orc.readOrcData`
+
+*Read an Arrow table from an Apache ORC file and convert to a kdb+ mixed list of array data*
+
+```txt
+.arrowkdb.orc.readOrcData[orc_file;options]
+```
+
+Where:
+
+- `orc_file` is a string containing the ORC file name
+- `options` is a kdb+ dictionary of options or generic null (`::`) to use defaults.  Dictionary key must be a `11h` list. Values list can be `7h`, `11h` or mixed list of `-7|-11|4|99|101h`.
+
+returns the array data
+
+> :warning: The Apache ORC file format is not supported on Windows platform and will return with the message **ORC files are not supported on Windows**.
+
+Supported options:
+
+- `ORC_CHUNK_SIZE` - Controls the approximate size of ORC data stripes within a column.  Long, default 1MB.
+
+```q
+q)i8_fd:.arrowkdb.fd.field[`int8;.arrowkdb.dt.int8[]];
+q)i16_fd:.arrowkdb.fd.field[`int16;.arrowkdb.dt.int16[]];
+q)i32_fd:.arrowkdb.fd.field[`int32;.arrowkdb.dt.int32[]];
+q)orc_schema:.arrowkdb.sc.schema[(i8_fd,i16_fd,i32_fd)];
+q)orc_data:(5?0x64;5?100h;5?100i);
+q).arrowkdb.orc.writeOrc["dataloader.orc";orc_schema;orc_data;options]
+q)read_data:.arrowkdb.orc.readOrcData["dataloader.orc";::]
+q)orc_data~read_data
+1b
+```
+
+### `orc.readOrcToTable`
+
+*Read an Arrow table from an Apache ORC file and convert to a kdb+ table*
+
+```txt
+.arrowkdb.orc.readOrcToTable[orc_file;options]
+```
+
+Where:
+
+- `orc_file` is a string containing the ORC file name
+- `options` is a kdb+ dictionary of options or generic null (`::`) to use defaults.  Dictionary key must be a `11h` list. Values list can be `7h`, `11h` or mixed list of `-7|-11|4|99|101h`.
+
+returns the kdb+ table
+
+> :warning: The Apache ORC file format is not supported on Windows platform and will return with the message **ORC files are not supported on Windows**.
+
+Each schema field name is used as the column name and the Arrow array data is used as the column data.
+
+Supported options:
+
+- `ORC_CHUNK_SIZE` - Controls the approximate size of ORC data stripes within a column.  Long, default 1MB.
+
```

Comment:
```suggestion
- `USE_MMAP` - Flag indicating whether the Parquet file should be memory mapped in.  This can improve performance on systems which support mmap.  Long, default: 0.
- `DECIMAL128_AS_DOUBLE` - Flag indicating whether to override the default type mapping for the Arrow decimal128 datatype and instead represent it as a double (9h).  Long, default 0.
- `NULL_MAPPING` - Sub-directory of null mapping datatypes and values.  See [here](null-mapping.md) for more details.
- `WITH_NULL_BITMAP` - Flag indicating whether to return the data values and the null bitmap as separate structures. See [here](null-bitmap.md) for more details. Long, default 0.

```